rfcs/0000-tsdb-dimensions.md

```diff
@@ -0,0 +1,240 @@
+# 0000: TSDB Dimensions
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **TBD** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+<!--
+Stage 1: If the changes include field additions or modifications, please create a folder titled as the RFC number under rfcs/text/. This will be where proposed schema changes as standalone YAML files or extended example mappings and larger source documents will go as the RFC is iterated upon.
+-->
+
+<!--
+Stage X: Provide a brief explanation of why the proposal is being marked as abandoned. This is useful context for anyone revisiting this proposal or considering similar changes later on.
+-->
+
+## Fields
+
+This RFC proposes the annotating of certain ecs fields as `dimension`. This change is proposed to take the advantage of using `TSDB` offerred by the elasticsearch without impacting the data injetion.  
+
+Annotating field as `dimension` is one of the important step in the process of TSDB adoption. Failing to annotate adequate number of fields as `dimension` when `TSDB` is enabled may lead to data loss. A large majority of fields that must be annotated as `dimension` fields are ecs fields. Presently, the Integration (Service Integration, Cloud Native, etc ) developers are expected to annotate ecs fields as `dimensions` in integration configuration. To avoid the duplicatation in configuration, minimize data loss probability, the RFC is proposed. `dimension` field takes two values - `true` and `false`.
```

Comment:
Is whether or not a field is annotated as a `dimension` something we should be defining at all within ECS? My initial opinion is leaning towards no, but I don't know very much about TSDB and the necessary configuration.

Different implementations should decide when to annotate or not based on the use cases involved. The way a field is defined in ECS is the way it should be implemented across all uses of the field. Is that the case for TSDB and `dimensions`? Should fields still be annotated for non-TSDB use cases?

Reply:
1. The set of fields that are proposed here (as part of RFC) are those fields which must be annotated across all use cases (all datastreams in all packages). 

In addition to these fields, datastreams  may or may not have additional fields that may need `dimension:true` annotation. An integration package developer is expected to annotate ONLY the additional fields , if any, after this proposed change is accepted and included in ecs. 

2. With `dimension: true` annotation  and the user choose not to enable TSDB, it does not alter or affect the behaviour of the integration package.